p1-staking-pool-dyn/src/lib.rs

```diff
@@ -150,15 +151,15 @@ impl Contract {
         let aid = env::predecessor_account_id();
         let mut vault = self.get_vault_or_default(&aid);
         assert!(
-            vault.staked > 0 && amount <= vault.staked + MIN_STAKE,
```

Comment:
why change thing that are working?
This was like that for this case:
- vault has 0.488373634354
- the user calls `unstake(0.5)` 
- the contract works unstaking 0.4883736
---
with your changes now the contract won't work and the user needs to specify exactly 0.488373634354 to close the vault.
MIN_STAKE means "nor the user nor us care for amounts lower than this"

Reply:
* if user has `0.488...` then calling `unstake(0.5)` is wrong: user doesn't have that amount and if we change the value then we do something different then what the transaction is saying. 
* user should call `close` if he doesn't want to put an amount

Reply:
user can't call "close" the UI is for Stake/Unstake